src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const currentSum = sum;
+
+      sum = 0;
+
+      return currentSum;
+    }
+
+    sum += num;
```

Comment:
The addition operation is correct and it follows the task's requirement to accumulate the sum until `adder` is called without arguments.